### ISSUE TITLE
Enforce type for children of Tabs component

### ISSUE THREAD

**Issue by jonkoops** [patternfly/patternfly-react#6766]
The `Tabs` component can only accept children of the type of the `Tab` component. This is [documented by a comment](https://github.com/patternfly/patternfly-react/blob/004cdf8648be921f15c9dafa489964f4bc4fa8d5/packages/react-core/src/components/Tabs/Tabs.tsx#L22-L23), but not enforced by TypeScript. To resolve this issue the correct type should be added so errors can be caught at compile time.

**Comment by nicolethoen** [comment]
The updates made in https://github.com/patternfly/patternfly-react/pull/6767 caused a type error in OpenShift
```
../console-app/src/components/user-preferences/UserPreferencePage.tsx:137:17 - error TS2322: Type 'ReactElement<TabProps, string | JSXElementConstructor<any>>[]' is not assignable to type '(TabsChild | TabsChild[]) & ReactNode'.
  Type 'ReactElement<TabProps, string | JSXElementConstructor<any>>[]' is not assignable to type 'TabsChild[]'.
    Type 'ReactElement<TabProps, string | JSXElementConstructor<any>>' is not assignable to type 'TabsChild'.
      Type 'ReactElement<TabProps, string | JSXElementConstructor<any>>' is not assignable to type 'TabElement'.
        Type 'string | JSXElementConstructor<any>' is not assignable to type 'JSXElementConstructor<TabProps>'.
          Type 'string' is not assignable to type 'JSXElementConstructor<TabProps>'.

137                 {userPreferenceTabs}
                    ~~~~~~~~~~~~~~~~~~~~

  ../../node_modules/@patternfly/react-core/dist/esm/components/Tabs/Tabs.d.ts:14:5
    14     children: TabsChild | TabsChild[];
           ~~~~~~~~
    The expected type comes from property 'children' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Tabs> & Pick<Readonly<TabsProps> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'
```

**Comment by jonkoops** [comment]
@nicolethoen Interesting, does the code there actually work? Seem like they are using strings as children where they would not be possible.

**Comment by nicolethoen** [comment]
@tlabaj do we want to reopen the associated PR for this issue and merge it into the v5 branch?

**Comment by jonkoops** [comment]
@nicolethoen I've re-opened this PR targeting the `v5` branch instead.

**Comment by nicolethoen** [comment]
Closed by #8217
